cpp/src/barretenberg/dsl/acir_format/sha256_constraint.hpp

```diff
@@ -10,13 +10,17 @@ struct Sha256Input {
     uint32_t num_bits;
 
     friend bool operator==(Sha256Input const& lhs, Sha256Input const& rhs) = default;
+    // for serialization, update with any new fields
```

Comment:
taking feedback on whether this is better as a tight MSGPACK
or
void msgpack(auto archive) {
   archive(NVP(fields...));
}
NVP is needed though to get field names, at a minimum

Reply:
I do like the word MSGPACK in there as I feel it needs some good correlation with the error stack traces people might face
MSGPACK_FIELDS isn't bad, I'll consider it